packages/cli/src/commands/build.ts

```diff
@@ -608,6 +650,33 @@ async function doBuild(
   );
 }
 
+export async function detectFrameworkVersion(
+  frameworkRecord: Framework,
+  output: Output
+) {
+  if (!frameworkRecord?.versionDependencies?.length) {
+    return;
+  }
+
+  for (let packageName of frameworkRecord.versionDependencies) {
+    const version = lookupInstalledVersion(packageName, output);
+    if (version) {
+      return {
+        version,
+      };
+    }
+  }
+}
+
+function lookupInstalledVersion(packageName: string, output: Output) {
+  try {
+    const pkgJson = require(`${packageName}/package.json`) as PackageJson;
+    return pkgJson.version;
```

Comment:
Using `require()` causes the following error to be thrown:

```
Error during lookupInstalledVersion: Error: Cannot find module 'gatsby/package.json'
```

The culprit is `ncc` because it resolves `'gatsby/package.json'` to the module index in the bundle, not the actual file.

There's probably multiple ways to solve this, but here's a one liner that will work:

```suggestion
return spawnSync(process.execPath, [
      '-p',
      `require('${packageName}/package.json').version`,
    ], {
      encoding: 'utf-8',
    }).stdout.trim();
```

Reply:
`eval('require')`

Reply:
Good catch! I forgot about that.

Reply:
Updated in [`057d0ed` (#9009)](https://github.com/vercel/vercel/pull/9009/commits/057d0ed269a14b272266b292c8bc8776496a5593)